### PR TITLE
Remove the APIs that were deprecated for removal in 4.2

### PR DESCRIPTION
**Pull request by chrisvest**
Motivation:
These APIs were deprecated with a message saying they would disappear in 4.2.

Modification:
Remove the APIs as promised.
All except ResourceLeakException were internal APIs.

All deprecated APIs that were NOT specifically promised to be removed in 4.2 have been left alone.

Result:
Slightly fewer deprecated APIs.

### CODE REVIEW

**Review by normanmaurer — APPROVED** [review]

File: common/src/main/java/io/netty/util/ResourceLeakException.java [hunk 1]
```diff
@@ -1,70 +0,0 @@
...
-
-package io.netty.util;
-
-import java.util.Arrays;
-
-/**
- * @deprecated This class will be removed in the future version.
- */
-@Deprecated
-public class ResourceLeakException extends RuntimeException {
```
Comment (idelpivnitskiy) [inline common/src/main/java/io/netty/util/ResourceLeakException.java:25]: My 2c: all removals in this PR does not seem like too much of the tech debt and should be ok to carry forward for all 4.x until 5.x. With any removal (even if they are in an "internal" package), you increase the risk of either a new version not being adopted or creating a runtime issues for users (imaging one dependency that upgraded to netty 4.2 and another one that is still on 4.1, it runtime they may not work with each other). 

I would propose to use 4.2 only to rise the min JDK version, while keeping all API as backward compatible.

If you really wanna try removing them, let's try. Just be ready to revert them back in the following 4.2.x versions if it causes issue for users. Taking into account how netty is wide-spread, the likelihood for that to happen is quite high.

File: handler/src/main/java/io/netty/handler/ipfilter/UniqueIpFilter.java [hunk 2]
```diff
@@ -33,7 +33,7 @@
 @ChannelHandler.Sharable
 public class UniqueIpFilter extends AbstractRemoteAddressFilter<InetSocketAddress> {
 
-    private final Set<InetAddress> connected = new ConcurrentSet<InetAddress>();
+    private final Set<InetAddress> connected = ConcurrentHashMap.newKeySet();
```
Comment (idelpivnitskiy) [inline handler/src/main/java/io/netty/handler/ipfilter/UniqueIpFilter.java:36]: It's ok to make internal changes like this one

File: common/src/main/java/io/netty/util/ResourceLeakException.java (hunk 1 above)
Comment (chrisvest) [reply common/src/main/java/io/netty/util/ResourceLeakException.java:25]: Code search on GitHub finds no usages of this exception, outside of Netty forks.

The IntegerHolder has a few usages: https://github.com/search?q=%22import+io.netty.util.internal.IntegerHolder%22&type=code

Unfortunately, there has been a fair amount of appetite for the ConcurrentSet: https://github.com/search?q=%22import+io.netty.util.internal.ConcurrentSet%22&type=code

So if we're having second thoughts, then the ConcurrentSet is the most important of the classes to keep.
What do you say, @normanmaurer ?

**Review by bryce-anderson — COMMENTED** [review]
my 2 cents: I'm fine with removing the API's, they seem unlikely to be used and that is part of what minor version bumps are good for imo. We can also add them back if we need to: nobody says the first release has to work perfectly but I don't think we can remove deprecations (at least per the guidelines) after we get the first cut published.

**Comment by normanmaurer** [comment]
Maybe we should just not do it to minimize the risk as @idelpivnitskiy mentioned. It does not harm us a lot to keep these around a bit more

**Comment by idelpivnitskiy** [comment]
Effort of going back and forth seems higher to me (plus it affects relationships with customers) vs just keeping API as-is.

**Comment by chrisvest** [comment]
Let's close this.

**Pull request closed without merging**